pydra/engine/specs.py

```diff
@@ -50,6 +58,139 @@ def retrieve_values(self, wf, state_index=None):
         for field, value in temp_values.items():
             setattr(self, field, value)
 
+    def check_metadata(self):
```

Comment:
check_metadata could take a flag that can add supported_keys to a base list.

Reply:
you mean an option to add something to the list defined in the function